src/lib/Wst/Server.hs

```diff
@@ -130,7 +131,8 @@ queryBlacklistedNodes _ (SerialiseAddress addr) = do
         . P.fromBuiltin
         . blnKey
         . uDatum
-  Env.withEnv $ Env.withTransfer transferLogic (fmap (fmap getHash) (Query.blacklistNodes @era))
+      nonHeadNodes (P.fromBuiltin . blnKey . uDatum -> hsh) = hsh /= ""
+  Env.withEnv $ Env.withTransfer transferLogic (fmap getHash . filter nonHeadNodes <$> (Query.blacklistNodes @era))
```

Comment:
Desezialization would fail in the case of the head node with an empty payment credential